File: classify-server/src/main/java/io/ermdev/alice/repository/SubjectRepository.java
package io.ermdev.alice.repository;

import io.ermdev.alice.entity.Subject;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface SubjectRepository extends JpaRepository<Subject, Long> {

    @Query("from Subject where id=:subjectId")
    Subject findById(@Param("subjectId") Long subjectId);
}

File: classify-rest-webservice/src/main/java/io/classify/data/repository/SubjectRepository.kt
package io.classify.data.repository

import io.classify.data.entity.Subject
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface SubjectRepository : JpaRepository<Subject, Long> {

    @Query("from Subject where id=:subjectId")
    fun findById(@Param("subjectId") subjectId: Long): Subject

    @Query("from Subject")
    override fun findAll(): List<Subject>
}